### ISSUE TITLE
Brendon JOSS review items

### ISSUE THREAD

**Issue by brendonhall** [rgmyr/corebreakout#6]
Hi all, I've begun reviewing for openjournals/joss-reviews#1969.

I was able to install the library I think.  I've have been trying to use the library to train a model.  The documentation on how to do this is quite brief.  I think you really need an example of a typical labeling workflow, and demonstrate how to use `labelme` to do this.  Some specific items:

- what class names are needed to identify the core sticks?  I was able to see a background class, try and columns by displaying a PolygonDataset() object, but no other info about them.

- there isn't a good description of the relationship between classes and instances.  Are tray objects all one instance?  is background?  presumably core columns are not?

- how much training data is needed?

- the `train_mrcnn_model.ipynb` notebook refers to a pretrained model.  This isn't in the repo, and a didn't see a reference about where to get it

- the default directories `assets/*` don't exist

- where is the provided BGS model and data?

**Comment by zanejobe** [comment]
Hi Brendon, 

Thanks for taking the time to review! 

I'll let Ross or Thomas answer some of the other questions, but the last three bullet points I can answer quickly. Go here https://github.com/rgmyr/corebreakout/releases and download `assets`.  There are two folders in there, one which contains the data (test and train) and another folder with two pre-trained models.

**Comment by ThomasMGeo** [comment]
There are also some issues with wholesale downloading all of the BGS photo's and re-hosting them from a copyright perspective. We used 30-50 boxes, with limited uplift towards 50.

**Comment by rgmyr** [comment]
Hi Brendon,

- Regarding the `assets`, we do explain where to find those in the [Getting Started section](https://github.com/rgmyr/corebreakout#download-data-optional) of the README (just above Installation). If you have suggestions about how to make that more clear/obvious, I'd be happy to update.
- I think your questions about labeled classes/instances are answered in [docs/creating_datasets.md](https://github.com/rgmyr/corebreakout/blob/master/docs/creating_datasets.md). Let me know if you've seen that... if it's not clear enough, I can add something that's more explicit about instances vs. classes. We could also add some labeled screenshots if that would help.

**Comment by ThomasMGeo** [comment]
Hi @brendonhall , the documentation just was updated, give it another look if you have a chance.

**Comment by rgmyr** [comment]
Hi @brendonhall, I'm just going to use this thread to address some of your other recent comments.

I'm going to try and pull together something that's a bit more out-of-the-box, but still with good coverage of the functionality. It's a little harder to do that with `select_model.ipynb` in particular, because that notebook presupposes that you have a folder full of model checkpoints, which I'm reluctant to provide since it would increase the size of `assets.zip` by a lot. At the very least I'll add some markdown to the notebook to do a better job of explaining the usage -- it should be fairly usable *after* you've done a training run.

In any case, I have added a new notebook called `inspect_dataset.ipynb`, which, as you might expect, inspects the small dataset in `tests/data` (which has all the same general properties as the datasets in `assets`). This is tied together with an expansion of the documentation on [Creating Datasets](https://corebreakout.readthedocs.io/en/latest/creating_datasets.html). I may yet flesh that out or reorganize a bit more, but at the very least I hope it's more clear about how to use `labelme`, and the distinction between classes/instances.

**Comment by rgmyr** [comment]
I've just pushed another commit which adds some markdown instructions and comments to `select_model.ipynb`.

I've also added a new notebook called `segmenter_demo.ipynb`, which demos the segmentation API and should work "out of the box".

**Comment by brendonhall** [comment]
Hi @rgmyr, this thread is fine and thanks for getting back to me.  And yeah, my comments a while ago were similar.  When I start using a new library, I basically look for how to get started as quickly as possible.  Usually a tutorial, that I can get working and then start modifying that for my specific problem. 

`inspect_dataset.ipynb` is very useful in this regard.  For example pointing out that the background class is implicitly assigned.  This does mean though that you must label ALL non-background objects in an image, correct?

I appreciate you attention to the comments that I've made.  This is a useful tool for researchers as well as having industrial applicaitons (energy, mining)

**Comment by rgmyr** [comment]
> This does mean though that you must label ALL non-background objects in an image, correct?

Yes, but only the objects that need to be detected because they have some purpose in processing (either core columns or "measuring stick" type objects that you intend to actually use as such). The background doesn't need to be visually uniform, and is simply a catch-all for everything that's not directly useful.

I'm going to close this issue so that Katy can go ahead with finalizing the review, but feel free to open another issue if you use this project in the future and run into any problems.

Thanks for your time and constructive feedback!
